src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1626,6 +1642,55 @@ static bool CanCombineCounts(int nodeMin, int nodeMax, int nextMin, int nextMax)
                             }
                             break;
 
+                        // Coalescing a loop with a subsequent string
+                        case RegexNodeKind.Oneloop or RegexNodeKind.Oneloopatomic or RegexNodeKind.Onelazy when nextNode.Kind == RegexNodeKind.Multi && currentNode.Ch == nextNode.Str![0]:
```

Comment:
Should this optimization also happen when nextNode.Kind == One and the character matches?

Reply:
Do you mean like https://github.com/dotnet/runtime/pull/65273/files/209eef017459fe2f2c6a697b62d4d2d31ef8c76a#diff-69d0a75aaf0674d30e057b6ac1d37203ee4757db089bac19f32a1d31d52a10c9R1629-R1630 ?

Reply:
oh, I missed that, yeah that is what I meant.